### PR TITLE
refactor: Use single settings.gradle for flank projects

### LINKED ISSUE #1112: Refactor Flank repo to single setting.gradle
**Author the user story for this feature**

As a developer, I want to work on a single Gradle project with multiple modules instead of multiple projects so I can edit and build everything in a single IDE instance.

**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I need to open a new instance of IDE or switch to another when I want to edit a different project from flank repository.

**Describe the solution you'd like**
All Gradle projects from the Flank's repository should be converted to Gradle modules and included in a single `setting.gradle.kt`

**Additional context**
This issue may impact have an impact on #1095 and #1109

### PR DESCRIPTION
**Pull request by jan-gogo**
Fixes #1112 

All github actions should work.

### CODE REVIEW

**Review by piotradamczyk5 — APPROVED** [review]

**Review by adamfilipow92 — APPROVED** [review]
Great work 👍

**Review by axelzuziak-gogo — APPROVED** [review]

File: buildSrc/src/main/kotlin/Deps.kt [hunk 1]
```diff
@@ -83,6 +83,14 @@ object Versions {
 
     // https://github.com/ben-manes/gradle-versions-plugin/releases
     const val BEN_MANES = "0.28.0"
+
+
+    // ============== flank-scripts ==============
+    const val KOTLIN_VERSION = "1.3.72"
```
Comment (Sloox) [inline buildSrc/src/main/kotlin/Deps.kt:89]: Kotlin 1.4?

File: flank-scripts/src/test/kotlin/flank/scripts/ci/nexttag/NextReleaseTagGeneratorTest.kt [hunk 2]
```diff
@@ -23,4 +24,88 @@ class NextReleaseTagGeneratorTest {
...
+                mockkStatic(LocalDate::class) {
+                    // Mockk probably has a bug because sometimes first call of
+                    // every { LocalDate.now() } is failing with strange error:
+                    /*
+                  every/verify {} block were run several times. Recorded calls count differ between runs
+Round 1: class java.time.LocalDate.of(-999999999, 1, 1), class java.time.LocalDate.of(999999999, 12, 31), class java.time.LocalDate.now()
+Round 2: class java.time.LocalDate.now()
+io.mockk.MockKException: every/verify {} block were run several times. Recorded calls count differ between runs
+Round 1: class java.time.LocalDate.of(-999999999, 1, 1), class java.time.LocalDate.of(999999999, 12, 31), class java.time.LocalDate.now()
+Round 2: class java.time.LocalDate.now()
```
Comment (Sloox) [inline flank-scripts/src/test/kotlin/flank/scripts/ci/nexttag/NextReleaseTagGeneratorTest.kt:42]: do we explicitly need to have the entire stacktrace here taking up 40+ lines? Possibly just a `...`for the less important aspects would suffice.

**Comment by Sloox** [comment]
The changes here will affect the incoming windows workflow #770 . which one should be merged first?

**Comment by piotradamczyk5** [comment]
> The changes here will affect the incoming windows workflow #770 . which one should be merged first?

I think that this PR should be merged first, however Janek is off until Friday, maybe someone could take care of it?

File: buildSrc/src/main/kotlin/Deps.kt (hunk 1 above)
Comment (piotradamczyk5) [reply buildSrc/src/main/kotlin/Deps.kt:89]: FIXED

File: flank-scripts/src/test/kotlin/flank/scripts/ci/nexttag/NextReleaseTagGeneratorTest.kt (hunk 2 above)
Comment (piotradamczyk5) [reply flank-scripts/src/test/kotlin/flank/scripts/ci/nexttag/NextReleaseTagGeneratorTest.kt:42]: FIXED

**Review by Sloox — APPROVED** [review]

**Comment by jan-gogo** [comment]
> I think that this PR should be merged first, however Janek is off until Friday

Feel free to finish it without me ;).

**Comment by piotradamczyk5** [comment]
> > I think that this PR should be merged first, however Janek is off until Friday
> 
> Feel free to finish it without me ;).

I will, not worries :D

**Comment by adamfilipow92** [comment]
Integration tests fixed

**Pull request merged**